app/services/group_service.rb

```diff
@@ -41,6 +41,7 @@ def self.update(group:, params:, actor:)
 
   def self.archive(group:, actor:)
     actor.ability.authorize! :archive, group
+    SubscriptionService.new(group, actor).end_subscription!
```

Comment:
I think, from memory, it was because the memberships are archived therefore you don't have permission to change the subscription...or something like that